In [16]:
 import os
 os.getcwd()

'C:\\Users\\aalia\\Downloads\\COMP SCI\\notebook'

In [17]:
import pandas as pd

In [19]:
df = pd.read_csv("data/titanic_data.csv")

df

survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
0           0       3    male  22.0      1      0   7.2500        S   Third   
1           1       1  female  38.0      1      0  71.2833        C   First   
2           1       3  female  26.0      0      0   7.9250        S   Third   
3           1       1  female  35.0      1      0  53.1000        S   First   
4           0       3    male  35.0      0      0   8.0500        S   Third   
..        ...     ...     ...   ...    ...    ...      ...      ...     ...   
884         0       2    male  27.0      0      0  13.0000        S  Second   
885         1       1  female  19.0      0      0  30.0000        S   First   
886         0       3  female   NaN      1      2  23.4500        S   Third   
887         1       1    male  26.0      0      0  30.0000        C   First   
888         0       3    male  32.0      0      0   7.7500        Q   Third   

       who  adult_male deck  embark_town alive  alone  
0      man        True  NaN  Southampton    no  False  
1    woman       False    C    Cherbourg   yes  False  
2    woman       False  NaN  Southampton   yes   True  
3    woman       False    C  Southampton   yes  False  
4      man        True  NaN  Southampton    no   True  
..     ...         ...  ...          ...   ...    ...  
884    man        True  NaN  Southampton    no   True  
885  woman       False    B  Southampton   yes   True  
886  woman       False  NaN  Southampton    no  False  
887    man        True    C    Cherbourg   yes   True  
888    man        True  NaN   Queenstown    no   True  

[889 rows x 15 columns]

In [29]:
df.dropna()

survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
1           1       1  female  38.0      1      0  71.2833        C  First   
3           1       1  female  35.0      1      0  53.1000        S  First   
6           0       1    male  54.0      0      0  51.8625        S  First   
10          1       3  female   4.0      1      1  16.7000        S  Third   
11          1       1  female  58.0      0      0  26.5500        S  First   
..        ...     ...     ...   ...    ...    ...      ...      ...    ...   
869         1       1  female  47.0      1      1  52.5542        S  First   
870         0       1    male  33.0      0      0   5.0000        S  First   
877         1       1  female  56.0      0      1  83.1583        C  First   
885         1       1  female  19.0      0      0  30.0000        S  First   
887         1       1    male  26.0      0      0  30.0000        C  First   

       who  adult_male deck  embark_town alive  alone  
1    woman       False    C    Cherbourg   yes  False  
3    woman       False    C  Southampton   yes  False  
6      man        True    E  Southampton    no   True  
10   child       False    G  Southampton   yes  False  
11   woman       False    C  Southampton   yes   True  
..     ...         ...  ...          ...   ...    ...  
869  woman       False    D  Southampton   yes  False  
870    man        True    B  Southampton    no   True  
877  woman       False    C    Cherbourg   yes  False  
885  woman       False    B  Southampton   yes   True  
887    man        True    C    Cherbourg   yes   True  

[182 rows x 15 columns]

In [22]:
df.columns

Index(['survived', 'pclass', 'sex', 'age', 'sibsp', 'parch', 'fare',
       'embarked', 'class', 'who', 'adult_male', 'deck', 'embark_town',
       'alive', 'alone'],
      dtype='object')

dimensions are 889 x 15 collumns

In [24]:
df.groupby

<bound method DataFrame.groupby of      survived  pclass     sex   age  sibsp  parch     fare embarked   class  \
0           0       3    male  22.0      1      0   7.2500        S   Third   
1           1       1  female  38.0      1      0  71.2833        C   First   
2           1       3  female  26.0      0      0   7.9250        S   Third   
3           1       1  female  35.0      1      0  53.1000        S   First   
4           0       3    male  35.0      0      0   8.0500        S   Third   
..        ...     ...     ...   ...    ...    ...      ...      ...     ...   
884         0       2    male  27.0      0      0  13.0000        S  Second   
885         1       1  female  19.0      0      0  30.0000        S   First   
886         0       3  female   NaN      1      2  23.4500        S   Third   
887         1       1    male  26.0      0      0  30.0000        C   First   
888         0       3    male  32.0      0      0   7.7500        Q   Third   

       who  adul

In [32]:
df.groupby(['class']).sum()

C:\Users\aalia\AppData\Local\Temp\ipykernel_21756\624196825.py:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  df.groupby(['class']).sum()


survived  pclass      age  sibsp  parch        fare  adult_male  alone
class                                                                         
First        136     216  7111.42     90     77  18177.4125         119    109
Second        87     368  5168.83     74     70   3801.8417          99    104
Third        119    1467  8894.92    301    193   6699.0493         317    323

in First class 136 people survived, 87 people survived in second, 119 in third.

In [34]:
df.groupby(['embark_town']).sum()

C:\Users\aalia\AppData\Local\Temp\ipykernel_21756\2290555382.py:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  df.groupby(['embark_town']).sum()


survived  pclass       age  sibsp  parch        fare  adult_male  \
embark_town                                                                     
Cherbourg          93     317   4005.92     65     61  10072.2962          90   
Queenstown         30     221    786.50     32     13   1014.5043          36   
Southampton       217    1511  16282.75    368    266  17431.5030         409   

             alone  
embark_town         
Cherbourg       85  
Queenstown      57  
Southampton    392

 the average fare for each passenger 10072.2962, 1014.5043, 17431.5030

In [35]:
df.groupby(['who']).sum()

C:\Users\aalia\AppData\Local\Temp\ipykernel_21756\1019958954.py:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  df.groupby(['who']).sum()


survived  pclass       age  sibsp  parch        fare  adult_male  alone
who                                                                           
child        49     218    528.67    144    105   2721.2210           0      6
man          88    1268  13670.50    158     82  13336.4198         535    409
woman       205     565   6976.00    163    153  12620.6627           0    121

For males its 13336.4198, for females its 12620.6627. 

In [41]:
df.groupby(['age' , 'fare' , 'embark_town']).sum()

C:\Users\aalia\AppData\Local\Temp\ipykernel_21756\84497533.py:1: FutureWarning: The default value of numeric_only in DataFrameGroupBy.sum is deprecated. In a future version, numeric_only will default to False. Either specify numeric_only or select only columns which should be valid for the function.
  df.groupby(['age' , 'fare' , 'embark_town']).sum()


survived  pclass  sibsp  parch  adult_male  alone
age   fare    embark_town                                                   
0.42  8.5167  Cherbourg           1       3      0      1           0      0
0.67  14.5000 Southampton         1       2      1      1           0      0
0.75  19.2583 Cherbourg           2       6      4      2           0      0
0.83  18.7500 Southampton         1       2      1      1           0      0
      29.0000 Southampton         1       2      0      2           0      0
...                             ...     ...    ...    ...         ...    ...
70.50 7.7500  Queenstown          0       3      0      0           1      1
71.00 34.6542 Cherbourg           0       1      0      0           1      1
      49.5042 Cherbourg           0       1      0      0           1      1
74.00 7.7750  Southampton         0       3      0      0           1      1
80.00 30.0000 Southampton         1       1      0      0           1      1

[637 rows x 6 columns]